Objects/tupleobject.c

```diff
@@ -83,6 +83,21 @@ PyTuple_New(Py_ssize_t size)
     return (PyObject *) op;
 }
 
+PyObject *
+_PyTuple_New_Nonzeroed(Py_ssize_t size)
+{
+    PyTupleObject *op;
+    if (size == 0) {
+        return tuple_get_empty();
+    }
+    op = tuple_alloc(size);
+    if (op == NULL) {
+        return NULL;
+    }
+    _PyObject_GC_TRACK(op);
```

Comment:
That's a bad practice: uninitialized tuples must be not tracked by the GC, it should only be tracked once all items are initialized. See https://github.com/python/cpython/issues/59313 I added _PyType_AllocNoTrack() to attempt to fix a similar issue in a few functions.

PyTuple_Pack() doesn't have this design flaw.